### PR TITLE
Change list of menus and add hover effect of menu

### LINKED ISSUE #179: ui: update header style
1. Update the Header style to fit the new look.

image

2. All links have hovers

image

3. The title should be passed from the config.
5. Links list should be passed from the config as well 

Figma: https://www.figma.com/file/P3X5qbxV5DZ0cM2arsmkrP/docs?node-id=122%3A1675

### PR DESCRIPTION
**Pull request by Y-k-Y**
Resolves #179

### CODE REVIEW

**Review by robonetphy — CHANGES_REQUESTED** [review]
image

@Y-k-Y 
As you can see in the above image.
heading's font size and padding as well as the header height are not the same as mentioned in Figma. Pls try to make pixel perfect as mentioned in Figma

File: src/frontend/styles/components/header.pcss [hunk 1]
```diff
@@ -63,9 +64,12 @@
 
     a:not(.docs-header__button) {
       color: inherit;
+      padding: 8px;
+      border-radius: 8px;
+      transition: background-color .3s;
 
       &:hover {
-        color: var(--color-link-active);
+        background-color: #F3F6F8;
```
[the remark this answers is not in the document]
Comment (Y-k-Y) [reply src/frontend/styles/components/header.pcss]: Created new var named `--color-link-hover` but not sure if this name is ok since the same color is used in a lot of other elements..

**Comment by talyguryn** [comment]
something is wrong with button style

image

also there are too slow animations. do we need them?

**Comment by talyguryn** [comment]
ok, button size is good now. but what is wrong with the header height?

image

**Comment by talyguryn** [comment]
need to ask design department to check it

File: src/frontend/styles/components/header.pcss [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
   display: flex;
+  justify-content: space-between;
   flex-shrink: 0;
-  padding: 0 var(--layout-padding-horisontal);
+  padding: 8px 16px;
   border-bottom: 1px solid var(--color-line-gray);
-  font-size: 15.8px;
-  line-height: 50px;
+  font-size: 18px;
+  height: 57px;
```
Comment (robonetphy) [inline src/frontend/styles/components/header.pcss:8]:
```suggestion
```

File: src/frontend/styles/components/header.pcss [hunk 3]
```diff
@@ -15,6 +16,8 @@
   &__menu-link,
   &__logo {
     display: inline-block;
+    text-decoration: none;
+    align-self: center;
```
Comment (robonetphy) [inline src/frontend/styles/components/header.pcss:20]:
```suggestion
    align-self: center;
    padding: 10px 0px;
    line-height: 21px;
```

**Review by robonetphy — CHANGES_REQUESTED** [review]
@Y-k-Y, we can maintain height of the header by using `padding` and `line-height` CSS properties.
you can try the changes I mentioned.

File: src/frontend/styles/components/header.pcss [hunk 4]
```diff
@@ -15,6 +16,8 @@
...
   }
 
   &__logo {
```
Comment (robonetphy) [inline src/frontend/styles/components/header.pcss:23]:
```suggestion
  &__logo {
    padding: 10px 0px;
```

**Review by robonetphy — CHANGES_REQUESTED** [review]

File: src/frontend/styles/components/header.pcss (hunk 2 above)
Comment (robonetphy) [inline src/frontend/styles/components/header.pcss:8]:
```suggestion
```

**Review by robonetphy — CHANGES_REQUESTED** [review]

File: src/frontend/styles/components/header.pcss (hunk 2 above)
Comment (Y-k-Y) [reply src/frontend/styles/components/header.pcss]: Fixed

File: src/frontend/styles/components/header.pcss [hunk 5]
```diff
@@ -1,10 +1,11 @@
 .docs-header {
   display: flex;
+  justify-content: space-between;
   flex-shrink: 0;
-  padding: 0 var(--layout-padding-horisontal);
+  padding: 8px 16px;
```
[the remark this answers is not in the document]
Comment (Y-k-Y) [reply src/frontend/styles/components/header.pcss]: Fixed

**Comment by neSpecc** [comment]
Also, the text should be Medium

File: src/frontend/styles/vars.pcss [hunk 6]
```diff
@@ -22,13 +24,13 @@
   /**
    * Site layout sizes
    */
-  --layout-padding-horisontal: 30px;
+  --layout-padding-horizontal: 30px;
```
Comment (robonetphy) [inline src/frontend/styles/vars.pcss:27]:
```suggestion
  --layout-padding-horizontal: 16px;
```

Comment (Y-k-Y) [reply src/frontend/styles/vars.pcss]: Fixed

**Review by robonetphy — APPROVED** [review]

File: src/frontend/styles/vars.pcss [hunk 7]
```diff
@@ -3,6 +3,8 @@
   --color-text-second: #5d6068;
   --color-line-gray: #E8E8EB;
   --color-link-active: #2071cc;
+  --color-link-hover: #F3F6F8;
+  --color-button-danger: #ff5159;
```
Comment (neSpecc) [inline src/frontend/styles/vars.pcss:7]: looks like an irrelevant change

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: src/frontend/styles/vars.pcss (hunk 7 above)
Comment (robonetphy) [reply src/frontend/styles/vars.pcss:7]: Only `--color-button-danger` is irrelevant

Comment (Y-k-Y) [reply src/frontend/styles/vars.pcss]: Fixed

**Review by neSpecc — APPROVED** [review]
